### PR TITLE
Force kill testcluster nodes

### PR DESCRIPTION
**Pull request by alpar-t**
Since we are not reusing them, no point in waiting for a clean stop.

### CODE REVIEW

**Comment by atorok** [comment]
@elasticmachine  run the gradle build tests 1

File: buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java [hunk 1]
```diff
@@ -224,22 +224,30 @@ synchronized void stop(boolean tailLogs) {
...
+            processHandle.children().forEach(each -> stopHandle(each, forcibly));
         }
-        logProcessInfo("Terminating elasticsearch process:", processHandle.info());
+        logProcessInfo(
+            "Terminating elasticsearch process" + (forcibly ? " forcibly " : "gratefully") + ":",
+            processHandle.info()
+        );
         if (processHandle.isAlive()) {
-            processHandle.destroy();
+            if (forcibly) {
```
Comment (rjernst) [inline buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java]: I think the order of this all can be cleaned up to not have doubling up on waits.

**Review by rjernst — COMMENTED** [review]
Just one ask

File: buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java (hunk 1 above)
Comment (atorok) [reply buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java]: @rjernst done, ready for review

**Comment by atorok** [comment]
@elasticmachine run elasticsearch-ci/1 again

File: buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java (hunk 1 above)
Comment (rjernst) [reply buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java]: We're still doing multiple aliveness checks when we already know it is killed. Here is what I was envisioning:

```
if (processHandle.isAlive() == false) {
    logger.info("Process was not running when we tried to terminate it.");
	return;
}

// Stop all children first, ES could actually be a child when there's some wrapper process like on Windows.
processHandle.children().forEach(each -> stopHandle(each, forcibly));

logProcessInfo("Terminating elasticsearch process" + (forcibly ? " forcibly " : "gratefully") + ":", processHandle.info());

if (forcibly) {
    processHandle.destroyForcibly();
} else {
    processHandle.destroy();
    waitForProcessToExit(processHandle);
    if (processHandle.isAlive()) {
	    logger.info("process did not terminate after {} {}, stopping it forcefully",
	                       ES_DESTROY_TIMEOUT, ES_DESTROY_TIMEOUT_UNIT);
	    processHandle.destroyForcibly();
    }
}

waitForProcessToExit(processHandle);
if (processHandle.isAlive()) {
    throw new TestClustersException("Was not able to terminate elasticsearch process");
}
```

Comment (atorok) [reply buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java]: @rjernst implemented as suggested

File: buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java [hunk 2]
```diff
@@ -243,35 +243,45 @@ synchronized void stop(boolean tailLogs) {
...
-        if (processHandle.isAlive()) {
-            logger.info("process did not terminate after {} {}, stopping it forcefully",
-                ES_DESTROY_TIMEOUT, ES_DESTROY_TIMEOUT_UNIT
-            );
+
+        // Stop all children first, ES could actually be a child when there's some wrapper process like on Windows.
+        processHandle.children().forEach(each -> stopHandle(each, forcibly));
+
+        logProcessInfo(
+            "Terminating elasticsearch process" + (forcibly ? " forcibly " : "gratefully") + ":",
```
Comment (mark-vieira) [inline buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java]: Typo, should be "gracefully". Not say we are not grateful 😉

**Review by mark-vieira — APPROVED** [review]
Trivial typo. Otherwise LGTM.

File: buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java [hunk 3]
```diff
@@ -243,35 +243,45 @@ synchronized void stop(boolean tailLogs) {
...
+            processHandle.info()
+        );
+
+        if (forcibly) {
             processHandle.destroyForcibly();
+        } else {
+            processHandle.destroy();
+            waitForProcessToExit(processHandle);
+            if (processHandle.isAlive()) {
```
Comment (rjernst) [inline buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java]: We can invert this, so:

```
if (processHandle.isAlive() == false) {
  return
}
logger.info("process did not terminate after {} {}, stopping it forcefully",
                   ES_DESTROY_TIMEOUT, ES_DESTROY_TIMEOUT_UNIT);
processHandle.destroyForcibly();

**Review by rjernst — APPROVED** [review]
LGTM too. One last nit.

**Pull request merged**